Prepare for release (#538)
sentinel-hub/sentinelhub-py: CHANGELOG.MD

## [Version 3.10.3] - 2024-13-08

- Monitoring a batch job now concludes if the batch job is cancelled.

> - When monitoring a batch job it now waits for the status of the job to change (even if all tiles have completed) before finalization in order to return the correct status.

Reviewer: ```suggestion
- The monitoring function now waits for the status of the batch job to change (even if all tiles have completed) before finalization in order to return the correct status.
```
